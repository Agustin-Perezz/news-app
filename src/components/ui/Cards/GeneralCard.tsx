import { CardActionArea, CardContent, Typography, Box } from '@mui/material';
import { NewsProps } from '../../../types/news-props';
import { CategoriesArticles } from './Categories';
import { CustomCard, CustomChip } from './CustomStylesCard';
import { useLoadImage } from '../../../hooks';

import load_image from '../../../assets/images/load-image.jpg';

interface Props {
  dataArticle: NewsProps;
}

export const GeneralCard: React.FC<Props> = ({ dataArticle }) => {
  const { loaded, onError, onLoad } = useLoadImage();

  return (
    <CustomCard>
      <CardActionArea target="_blank" href={dataArticle.url} sx={{ height: '470px' }}>
        <Box sx={{ position: 'relative' }}>
          <img
            style={{ height: '185px' }}
            onLoad={onLoad}
            alt={dataArticle.nameSource}
            src={loaded ? dataArticle.image_url : load_image}
            onError={({ currentTarget }) => onError({ currentTarget })}
          />
          <CustomChip label={dataArticle.published_at} />
        </Box>
        <CardContent>
          <Typography data-testid="card-title" variant="subtitle1">
            {dataArticle.title}
          </Typography>
          <Typography data-testid="card-description" variant="body2" color="text.secondary">
            {dataArticle.description}
          </Typography>
          {dataArticle.categories && <CategoriesArticles listCategories={dataArticle.categories} />}
          {dataArticle.nameSource && (
            <Box sx={{ display: 'flex', marginTop: '10px' }}>
              <Typography variant="subtitle2" mr={1}>
                Source:
              </Typography>
              <Typography variant="body2" color="text.secondary" sx={{ lineHeight: '1.57' }}>
                {dataArticle.nameSource}
              </Typography>
            </Box>
          )}
        </CardContent>
      </CardActionArea>
    </CustomCard>
  );
};
