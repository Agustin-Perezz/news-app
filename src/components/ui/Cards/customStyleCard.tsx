import { styled, createTheme } from '@mui/material/styles';
import MuiCard from '@mui/material/Card';
import MuiChip from '@mui/material/Chip';

export const tabFont = createTheme({
  typography: {
    fontFamily: ['Heebo', 'sans-serif'].join(','),
  },
  palette: { secondary: { main: '#2475C7' } },
});

export const CustomCard = styled(MuiCard)(() => ({
  '&:hover': {
    boxShadow: '0px 2px 18px 0px rgba(0,0,0,0.8)',
    transform: 'scale(1.02)',
    transition: 'all .2s ease-in-out',
    cursor: 'pointer',
  },
  maxHeight: 'auto',
  minHeight: '447px',
  width: '278px',
}));

export const CustomChip = styled(MuiChip)(() => ({
  position: 'absolute',
  top: '8px',
  right: '5px',
  color: 'ghostwhite',
  backgroundColor: '#0A1929',
  opacity: '0.6',
  height: '29px',
  border: '1px solid #9393A5',
}));
