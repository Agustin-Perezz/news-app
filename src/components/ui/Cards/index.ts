export * from './ListCards';
export * from './customStyleCard';
