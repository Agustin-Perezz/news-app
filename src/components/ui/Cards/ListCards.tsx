import React from 'react';
import { Grid } from '@mui/material';
import { NewsProps } from '../../../types/news-props';
import { GeneralCard } from './GeneralCard';
import { RenderController } from '../RenderController';
import { CustomGrid, CustomTitle } from './CustomStyleCard';

interface Props {
  dataArticles?: NewsProps[];
  isLoading: boolean;
  isError?: string;
  titleText?: string;
}

export const ListCards: React.FC<Props> = ({
  dataArticles,
  isLoading,
  isError,
  titleText,
}) => {
  return (
    <RenderController
      isLoading={isLoading}
      isError={isError}
      data={dataArticles}
    >
      <Grid sx={{ flexGrow: 1, margin: '25px 0px' }}>
        <Grid
          item
          xs={12}
          sx={{
            justifyContent: 'center',
            display: 'flex',
            position: 'relative',
          }}
        >
          <CustomGrid container spacing={3}>
            {dataArticles?.map((article, index) => (
              <Grid key={article.uuid} item>
                <GeneralCard dataArticle={article} />
                {index === 0 && (
                  <CustomTitle variant="h6"> {titleText} </CustomTitle>
                )}
              </Grid>
            ))}
          </CustomGrid>
        </Grid>
      </Grid>
    </RenderController>
  );
};
