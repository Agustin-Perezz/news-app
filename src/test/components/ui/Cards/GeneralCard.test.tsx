import { render, screen } from '@testing-library/react';
import { GeneralCard } from '../../../../components/ui/Cards/GeneralCard';
import { mockArticle } from './mockArticles';

describe('Test in <GeneralCard />', () => {
  function setup() {
    return render(<GeneralCard dataArticle={mockArticle} />);
  }

  it('should match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('shuold render props correctly', () => {
    setup();
    screen.getByText(mockArticle.nameSource);
    screen.getByText(mockArticle.title);
    screen.getByText(mockArticle.description);
    screen.getByText(mockArticle.categories[0].type);
  });
});
