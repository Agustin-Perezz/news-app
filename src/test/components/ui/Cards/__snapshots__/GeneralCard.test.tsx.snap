// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GeneralCard /> should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-nkszem-MuiPaper-root-MuiCard-root"
    >
      <a
        class="MuiButtonBase-root MuiCardActionArea-root css-y29lt3-MuiButtonBase-root-MuiCardActionArea-root"
        href="https://google.com"
        tabindex="0"
        target="_blank"
      >
        <div
          class="MuiBox-root css-79elbk"
        >
          <img
            alt="calcionapoli24.it"
            src="[object Object]"
            style="height: 185px; min-width: 100%;"
          />
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-c8zb3x-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              11/25/2022
            </span>
          </div>
        </div>
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
            data-testid="card-title"
          >
            Omaggio Maradona a Fuorigrotta, c'Ã¨ anche Maradona junior | FOTO
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            data-testid="card-description"
          >
            Anche il figlio del Pibe presente stasera a Fuorigrotta
          </p>
          <div
            class="MuiBox-root css-3yoo1r"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-1ij1w03-MuiTypography-root"
            >
              Categories:
            </h6>
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-s89mk0-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                sports
              </span>
            </div>
          </div>
          <div
            class="MuiBox-root css-1hprwxc"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-1od5i09-MuiTypography-root"
            >
              Source:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2 css-bssea4-MuiTypography-root"
            >
              calcionapoli24.it
            </p>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</div>
`;
